contracts/helpers/SwapHelper.sol

```diff
@@ -64,11 +68,12 @@ library SwapHelper {
             uint256 amount1PostX96 = (ratioE18 * valueX96) / (ratioE18 + 1e18);
 
             token0In = !(amount1In >= (amount1PostX96 >> FixedPoint96.RESOLUTION));
+
             if (token0In) {
                 amountToSwap =
-                    (((amount1PostX96 - (amount1In << FixedPoint96.RESOLUTION)) / sqrtPriceX96) <<
+                    (((amount1PostX96 - (amount1In << FixedPoint96.RESOLUTION)) / uint256(sqrtPriceX96)) <<
```

Comment:
if we use safemath for uint256 we hav to call the operations with .sub and .div,...

Reply:
this was in another PR

Reply:
this was in another PR